<?xml version="1.0"?>
<launch>
  <arg name="device"    default="/dev/video0"/>
  <arg name="width"     default="640"/>
  <arg name="height"    default="480"/>
  <arg name="framerate" default="15"/>

  <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="screen">
    <param name="video_device"    value="$(arg device)"/>
    <param name="image_width"     value="$(arg width)"/>
    <param name="image_height"    value="$(arg height)"/>
    <param name="framerate"       value="$(arg framerate)"/>
    <param name="pixel_format"    value="yuyv"/>
    <param name="camera_frame_id" value="usb_cam"/>
    <param name="io_method"       value="mmap"/>

    <remap from="/usb_cam/image_raw" to="/image_in"/>
  </node>

  <node pkg="opengl_ros" type="color_extraction_cpu"
    name="color_extraction_cpu" output="screen">
    <param name="width"  value="$(arg width)"/>
    <param name="height" value="$(arg height)"/>

    <param name="threshold_l"   value="0"/>
    <param name="svm_coef_a"    value="0.266602955271"/>
    <param name="svm_coef_b"    value="0.0444656815907"/>
    <param name="svm_intercept" value="-44.3271851592"/>

    <remap from="image_in"  to="/image_in"/>
    <remap from="image_out" to="/image_out"/>
  </node>

  <node pkg="image_view" type="image_view" name="viewer_in">
    <remap from="image" to="/image_in"/>
  </node>

  <node pkg="image_view" type="image_view" name="viewer_out">
    <remap from="image" to="/image_out"/>
  </node>
</launch>
